R/as-tibble.R

```diff
@@ -1,26 +1,61 @@
+#' Flatten a Crunch Cube
+#'
+#' Crunch Cubes can be expressed as a long data frame instead of a
+#' multidimensional array. In this form each dimension of the cube is a variable
+#' and the cube values are expressed columns for each measure. This is useful
+#' both to better understand what each entry of a cube represents, and to work
+#' with the cube result using tidyverse tools.
+#'
+#' @param x a CrunchCube
+#' @param ... futher arguments passed on to `tibble::as_tibble()`
+#'
 #' @export
 #' @importFrom tibble as_tibble
+#' @importFrom crunch getDimTypes
+#' @importFrom dplyr bind_cols
+#' @importFrom purrr map2 map reduce
+#' @importFrom stringr str_extract
 as_tibble.CrunchCube <- function (x, ...) {
     ## TODO: Consider using `dplyr::tbl_cube` class
-    ## TODO: NA behavior? Add way to get missings not missing. (This uses
-    ## as.array to remove extraneous dims, and doing so applies the '@useNA'
-    ## from the cube)
-    ## TODO: deal with naming for array (and NUMR) variable dims (var.categories?)
-
-    ## 1) get the measures
-    ## TODO: better access methods for secondary measures in Cube
-    ## TODO: if weighted query, include ".unweighted_counts"
```

Comment:
Should the converse also be true: if the query isn't weighted, don't include .unweighted_counts? I think so; "unweighted" has no meaning when there are no weights, and if you wanted counts, you could ask for them.

Reply:
It seems safer to either include both columns or have an argument which prevents including `.unweighted_counts`. For instance if you relied on `.unweighted_counts` in a script, you wouldn't want that to fail if weighting were removed from the query. Also I like that `as_tibble` gives you everything that's in the cube, and I think we include unweighted counts whether in the cube regardless of whether the query is weighted.